common/reflectivity/reflectivity.cpp

```diff
@@ -0,0 +1,178 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include <numeric>
+#include "reflectivity.h"
+#include "types.h"
+
+using namespace rs2;
+
+
+static const int STD_PERIOD = 100;
+static const int NINETY_FIVE_PERCENT_OF_STD_PERIOD = static_cast<int>(0.95 * STD_PERIOD);
+static const int MAX_RANGE_IN_UNIT = 65536.0f;
+static const float LONG_THERMAL = 74.5f;
+static const float INDOOR_MAX_RANGE = 9.0f;
+
+//TODO try to read from LRS
+static const float FOV_H = 0.610865f;
+static const float FOV_V = 0.479966f;
+
+static const int VGA_HALF_WIDTH = 320;
+static const int VGA_HALF_HEIGHT = 240;
+
+static bool is_close_to_zero(float x) { return (std::abs(x) < std::numeric_limits<float>::epsilon()); }
+
+
+reflectivity::reflectivity() : _is_empty(false)
+{
+    _dist_queue.assign(STD_PERIOD, 0); // Allocate size for all samples in advance to minimize runtime.
+}
+
+//Reflectivity analysis is based on STD and normalized IR.Final reflectivity values are weighted average between them
+float reflectivity::get_reflectivity(float raw_noise_estimation,float max_usable_range, float ir_val) const
+{
+    //Calculating STD over time(temporal noise), when there are more than 5 % invalid return high value
+    std::vector<float> _filt_dist_arr(STD_PERIOD);
+
+    int count_valid = 0;
+
+    // Count non zero (or close to zero) elements in history
+    std::copy_if(_dist_queue.begin(), _dist_queue.end(), _filt_dist_arr.begin(),
+        [&count_valid]( float elem ) {
+        if (!is_close_to_zero(elem))
+        {
+            ++count_valid;
+            return true;
+        }
+        return false;
+    } );
+
+    if (0 == count_valid)
+        throw librealsense::invalid_value_exception("reflectivity N/A, not enough data was collected");
+
+    _filt_dist_arr.resize(count_valid); // resize to actual number of elements pushed
+
+    float sum = std::accumulate( _filt_dist_arr.begin(), _filt_dist_arr.end(), 0.0f );
+    float mean = sum / _filt_dist_arr.size();
+
+    float standard_deviation = 9999.0f;
+
+    if( count_valid >= NINETY_FIVE_PERCENT_OF_STD_PERIOD )
+    {
+        float variance = 0.0f;
+        for( auto val : _filt_dist_arr )
+            variance += pow( val - mean, 2.0f );
+
+        variance = variance / _filt_dist_arr.size();
+        standard_deviation = sqrt( variance );
```

Comment:
This is a `double` conversion, no? Maybe use `std::sqrt`...
Make sure no warnings.

Reply:
👍